### PR TITLE
Add Native Image Demo how to Build a Native Shared Library

### PR DESCRIPTION
**Pull request by LesiaChaban**
The demo originates from https://www.graalvm.org/latest/reference-manual/native-image/guides/build-native-shared-library/.

### CODE REVIEW

File: .github/workflows/native-shared-library.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    strategy:
+      matrix:
+        version: [latest, dev]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          version: ${{ matrix.version }}
+          java-version: '19'
+          components: 'native-image'
```
Comment (fniephaus) [inline .github/workflows/native-shared-library.yml:31]:
```suggestion
          components: 'native-image,llvm-toolchain'
```

and remove ` gu install llvm-toolchain` below.

File: native-shared-library/README [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#  Build a Native Shared Library
```
Comment (fniephaus) [inline native-shared-library/README:1]: Share libraries are always native, so this is probably better:

```suggestion
# Build a Shared Library with Native Image
```

File: native-shared-library/README [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+This demo shows how to create a small Java class library, use `native-image` to create a native shared library, and then create a small C application that uses that shared library.
+
+GraalVM makes it easy to use C to call into a native shared library.
+There are two primary mechanisms for calling a method (function) embedded in a native shared library: the [Native Image C API](https://www.graalvm.org/latest/reference-manual/native-image/native-code-interoperability/C-API/) and the [JNI Invocation API](https://www.graalvm.org/latest/reference-manual/native-image/native-code-interoperability/JNIInvocationAPI/).
+
+This examples demonstrates how to use the **Native Image C API**. You will:
```
Comment (fniephaus) [inline native-shared-library/README:8]:
```suggestion
This example demonstrates how to use the **Native Image C API**. We will:
```

File: native-shared-library/README [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    It will produce the following artifacts:
+    ```
+    --------------------------------------------------
+    Produced artifacts:
+    /demo/libenvmap.dylib (shared_lib)
+    /demo/libenvmap.h (header)
+    /demo/graal_isolate.h (header)
+    /demo/libenvmap_dynamic.h (header)
+    /demo/graal_isolate_dynamic.h (header)
+    /demo/libenvmap.build_artifacts.txt
```
Comment (fniephaus) [inline native-shared-library/README:49]: Remove this file:
```suggestion
    /demo/libenvmap.build_artifacts.txt
```

File: native-shared-library/README [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+    ```bash
+    native-image --shared -H:Name=<libraryname> <class name>
+    ```
+    ```bash
+    native-image --shared -jar <jarfile> -H:Name=<libraryname>
+    ```
+    Also, native shared library can be created from a JAR file using the following syntax:
+    ```bash
+    native-image -jar <jarfile> --shared
```
Comment (fniephaus) [inline native-shared-library/README:68]: This seems confusing, just remove it?
```suggestion
```

File: native-shared-library/README [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    ```   
+
+4. Compile the _main.c_ using `clang`.  
+    ```bash
+    $JAVA_HOME/languages/llvm/native/bin/clang -I ./ -L ./ -l envmap -Wl,-rpath ./ -o main main.c 
+    ```
+5. Run the C application by passing a string as an argument. For example:
```
Comment (fniephaus) [inline native-shared-library/README:75]:
```suggestion
5. And finally, run the C application by passing a string as an argument. For example:
```

**Review by fniephaus — APPROVED** [review]
LGTM. Please address my comments before merging, thanks!

File: native-shared-library/README [hunk 7]
```diff
@@ -0,0 +1,81 @@
+# Build a Shared Library with Native Image
```
Comment (olyagpl) [inline native-shared-library/README:1]: # Build a Native Shared Library with GraalVM Native Image

File: native-shared-library/README [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+This demo shows how to create a small Java class library, use `native-image` to create a native shared library, and then create a small C application that uses that shared library.
```
Comment (olyagpl) [inline native-shared-library/README:3]: @LesiaChaban, please move all content from the website guide https://www.graalvm.org/dev/reference-manual/native-image/guides/build-native-shared-library/. Add missing pieces in the introduction part.

**Review by olyagpl — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -78,6 +78,10 @@ cd graalvm-demos
     <tr>
       <td align="left" width="30%"><a href="/native-jfr-demo/">native-jfr-demo</a><br><a href="https://github.com/graalvm/graalvm-demos/actions/workflows/ni-native-executable-jfr-demo/.github/workflows/native-jfr-demo.yml"><img alt="native-jfr-demo" src="https://github.com/graalvm/graalvm-demos/actions/workflows/native-jfr-demo.yml/badge.svg"/></a></td>
       <td align="left" width="70%">Demonstrates how to create a custom JDK Flight Recorder (JFR) event and use that in a native executable<br><strong>Technologies: </strong> Native Image, JFR, VisualVM <br><strong>Reference: </strong><a href="https://www.graalvm.org/latest/reference-manual/native-image/guides/build-and-run-native-executable-with-jfr/">Build and Run Native Executables with JFR</a></td>
+    </tr>
+    <tr>
+      <td align="left" width="30%"><a href="/native-shared-library/">native-shared-library</a><br><a href="https://github.com/graalvm/graalvm-demos/actions/ni-shared-lib/.github/workflows/native-shared-library.yml"><img alt="native-shared-library" src="https://github.com/graalvm/graalvm-demos/actions/workflows/native-shared-library.yml/badge.svg"/></a></td>
```
Comment (olyagpl) [inline README.md:83]: https://github.com/graalvm/graalvm-demos/actions/ni-shared-lib/.github/workflows/native-shared-library.yml is not correct.

**Review by olyagpl — APPROVED** [review]

**Pull request merged**